### PR TITLE
texting hours per campaign

### PR DESCRIPTION
**Pull request by lperson**

### CODE REVIEW

**Comment by lperson** [comment]
closes #644

**Comment by shakalee14** [comment]
Awesome! @lperson - I'm having trouble getting this working locally / testing locally - let me know if these are the correct steps.

1) Run `npm run dev` - which logs that a migration has occurred on the campaign table
2) Create a campaign and add `texting hours`
screen shot 2018-09-03 at 12 29 53 pm
3) Go to organizational settings and add a different texting hours constraint
screen shot 2018-09-03 at 12 29 41 pm
4) Switch to texter - and it should respect texting hours set on the campaign - currently it's still greyed out: 
screen shot 2018-09-03 at 12 30 00 pm



Currently in the db the texting hours are set to what I set in the campaign and override_organization_texting_hours is set to true - but I'm not able to send texts.

**Comment by lperson** [comment]
@shakalee14 try refreshing the switch to texter view.

**Comment by lperson** [comment]
@shakalee14 what time is the machine set to, and in what timezone?

**Comment by lperson** [comment]
@shakalee14 also, please confirm you have the correct branch -- `new_texting_hours_per_campaign` **NOT** ~~texting_hours_per_campaign~~

**Comment by shakalee14** [comment]
Refreshed a few times with no luck - and my machine? It's in EST.

**Comment by lperson** [comment]
@shakalee14 also, do the contacts have zip codes?

**Comment by shakalee14** [comment]
I have the correct branch and I'm using this csv:
https://github.com/MoveOnOrg/Spoke/blob/main/__test__/test_data/female_scientists.csv

**Comment by lperson** [comment]
@shakalee14  the zip codes are in California.  Spoke is coded to respect the timezone of the zip code.  You set texting to start at noon, but it's about 9:48 in California, so none of those zipcodes are in a timezone where it's before noon.  The timezone in campaign texting hours config is used with contacts that have no zip code and to determine whether it's DST or not.  Try setting campaign texting hours to something before 9am.

File: src/components/CampaignTextingHoursForm.jsx [hunk 1]
```diff
@@ -0,0 +1,234 @@
+import type from 'prop-types'
+import Toggle from 'material-ui/Toggle'
+import React from 'react'
+import Form from 'react-formal'
+import GSForm from './forms/GSForm'
+import CampaignFormSectionHeading from './CampaignFormSectionHeading'
+import yup from 'yup'
+import { css, StyleSheet } from 'aphrodite'
```
Comment (shakalee14) [inline src/components/CampaignTextingHoursForm.jsx:8]: Can we removed this unused import.

File: src/components/CampaignTextingHoursForm.jsx [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+  }
+
+  addAutocompleteFormField(
+    name,
+    stateName,
+    initialValue,
+    label,
+    hint,
+    choices,
+    collection
```
Comment (shakalee14) [inline src/components/CampaignTextingHoursForm.jsx:60]: what is this used for? `collection` ?

Comment (lperson) [reply src/components/CampaignTextingHoursForm.jsx:60]: `choices` contains the strings to be displayed

`collection` is the original values, used by a callback created by this function, so when a choice is selected, we can find the value to include when we fire the selected event

File: src/components/CampaignTextingHoursForm.jsx (hunk 1 above)
Comment (lperson) [reply src/components/CampaignTextingHoursForm.jsx:8]: Sure!  ~~... but I think they're all used.~~

Comment (lperson) [reply src/components/CampaignTextingHoursForm.jsx:8]: Done.

**Comment by schuyler1d** [comment]
For future, I have resolved conflicts with this PR and `scaling-tip` in branch `new_texting_hours_per_campaign-withcaching`

**Comment by shakalee14** [comment]
Merged with #901  and #900

**Pull request closed without merging**